retention/options.go

```diff
@@ -64,12 +71,42 @@ func NewOptions() Options {
 		blockSize:                        defaultBlockSize,
 		bufferFuture:                     defaultBufferFuture,
 		bufferPast:                       defaultBufferPast,
-		bufferDrain:                      defaultBufferDrain,
 		dataExpiry:                       defaultDataExpiry,
 		dataExpiryAfterNotAccessedPeriod: defaultDataExpiryAfterNotAccessedPeriod,
 	}
 }
 
+func (o *options) Validate() error {
+	if o.bufferFuture < 0 {
+		return errBufferFutureNonNegative
+	}
+	if o.bufferPast < 0 {
+		return errBufferPastNonNegative
+	}
+	if o.blockSize <= 0 {
+		return errBlockSizePositive
+	}
+	if o.bufferFuture >= o.blockSize {
+		return errBufferFutureTooLarge
+	}
+	if o.bufferPast >= o.blockSize {
+		return errBufferPastTooLarge
+	}
+	if o.retentionPeriod < o.blockSize {
+		return errRetentionPeriodTooSmall
+	}
+	return nil
+}
+
+func (o *options) Equal(value Options) bool {
```

Comment:
NamespaceRegistry's can be specified in multiple places (e.g. storage Options and peer bootstrapper Options). So their validate methods registries (which in turn compare metadatas, which compare namespace options, which compare retention options).